src/runtime/pay-flow.js

```diff
@@ -349,12 +350,24 @@ export class PayCompleteFlow {
 
     let confirmFeUrl;
     if (args.productType === ProductType.VIRTUAL_GIFT) {
-      confirmFeUrl = feUrl('/payconfirmiframe', '', {
+      const urlParams = {
         productType: args.productType,
         publicationId: args.publicationId,
         offerId: this.sku_,
         origin: parseUrl(this.win_.location.href).origin,
-      });
+      };
+      if (response.requestMetadata) {
+        urlParams.canonicalUrl = response.requestMetadata.contentId;
+        urlParams.isAnonymous = response.requestMetadata.anonymous;
+      }
+      if (response.swgUserToken) {
+        urlParams.sut = response.swgUserToken;
+      }
+      const orderId = parseOrderIdFromPurchaseDataSafe(response.purchaseData);
+      if (orderId) {
+        urlParams.orderId = orderId;
+      }
+      confirmFeUrl = feUrl('/payconfirmiframe', '', urlParams);
```

Comment:
Any risk of these url params being copied ? Should these values be passed in web activities?